src/Client/ResourceClientInterface.php

```diff
@@ -88,9 +88,9 @@ public function upsertResource($uri, array $uriParameters = [], array $body = []
     /**
      * Updates or creates several resources.
      *
-     * @param string             $uri           URI of the resource
-     * @param array              $uriParameters URI parameters of the resource
-     * @param array|\Traversable $resources     array of resources to create or update
+     * @param string                $uri           URI of the resource
+     * @param array                 $uriParameters URI parameters of the resource
+     * @param array|StreamInterface $resources     Resources to create or update
```

Comment:
I would keep it like that `array of resources to create or update. You can pass your own StreamInterface implementation as well.`

Actually, array is the most common way to do it, so it's better to clearly document it. And it indicates that they must implement StreamInterface otherwise.